middleware/correlation.go

```diff
@@ -3,21 +3,26 @@ package middleware
 import (
 	"context"
 	"net/http"
+	"regexp"
 
 	cs "github.com/cloudtrust/common-service"
 	gen "github.com/cloudtrust/common-service/idgenerator"
 	"github.com/cloudtrust/common-service/log"
 	"github.com/cloudtrust/common-service/tracing"
 )
 
+const (
+	regExpCorrelationID = `^[\w\d_#@-]{1,70}$`
+)
+
 // MakeHTTPCorrelationIDMW retrieve the correlation ID from the HTTP header 'X-Correlation-ID'.
 // It there is no such header, it generates a correlation ID.
 func MakeHTTPCorrelationIDMW(idGenerator gen.IDGenerator, tracer tracing.OpentracingClient, logger log.Logger, componentName, componentID string) func(http.Handler) http.Handler {
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 			var correlationID = req.Header.Get("X-Correlation-ID")
 
-			if correlationID == "" {
+			if match, _ := regexp.MatchString(regExpCorrelationID, correlationID); !match {
```

Comment:
I am wondering if we should returns an error about invalid CorrelationId instead of fallback silently on this ?
What do you think ?

Reply:
New version pushed : Bad request is returned when correlation id is invalid